### ISSUE TITLE
Ignoring comments start at the beginning of the line

### ISSUE THREAD

**Issue by gerases** [erezsh/lark#40]
Hi,

First, thank you for all the work!

Here's my grammar:

```
      sudo_item : (alias | user_spec)*

      alias : "User_Alias"    user_alias  (":" user_alias)*
              | "Runas_Alias" runas_alias (":" runas_alias)*
              | "Host_Alias"  host_alias  (":" host_alias)*
              | "Cmnd_Alias"  cmnd_alias  (":" cmnd_alias)*

      user_alias  : ALIAS_NAME "=" user_list
      host_alias  : ALIAS_NAME "=" host_list
      runas_alias : ALIAS_NAME "=" runas_list
      cmnd_alias  : ALIAS_NAME "=" cmnd_list

      user_spec      : user_list host_list "=" cmnd_spec_list (":" host_list "=" cmnd_spec_list)*
      cmnd_spec_list : cmnd_spec | cmnd_spec "," cmnd_spec_list
      cmnd_spec      : runas_spec? tag_spec* COMMAND
      runas_spec     : "(" runas_list? (":" runas_list)? ")"

      host_list  : HOST_NAME ("," HOST_NAME)*
      user_list  : user      ("," user)*
      runas_list : USER_NAME ("," USER_NAME)*
      cmnd_list  : COMMAND   ("," COMMAND)*

      user :    "!"*                     USER_NAME
              | "!"* PERCENT             GROUP_NAME
              | "!"* HASH                UID
              | "!"* PERCENT_HASH        GID
              | "!"* PLUS                NETGROUP_NAME
              | "!"* PERCENT_COLON       NONUNIX_GROUP_NAME
              | "!"* PERCENT_COLON_HASH  NONUNIX_GID
              | "!"* user_alias

      tag_spec : (tag_nopwd
                  | tag_pwd
                  | tag_noexec
                  | tag_exec
                  | tag_setenv
                  | tag_nosetenv
                  | tag_log_output
                  | tag_nolog_output) ":"

      tag_pwd          : "PASSWD"
      tag_nopwd        : "NOPASSWD"
      tag_exec         : "EXEC"
      tag_noexec       : "NOEXEC"
      tag_setenv       : "SETENV"
      tag_nosetenv     : "NOSETENV"
      tag_log_output   : "LOG_OUTPUT"
      tag_nolog_output : "NOLOG_OUTPUT"

      PERCENT            : "%"
      HASH               : "#"
      PERCENT_HASH       : "%#"
      PLUS               : "+"
      PERCENT_COLON      : "%:"
      PERCENT_COLON_HASH : "%:#"
      UID                : /[0-9]+/
      GID                : /[0-9]+/
      NONUNIX_GID        : /[0-9]+/
      USER_NAME          : /[-_.a-z0-9A-Z]+/
      GROUP_NAME         : CNAME
      NETGROUP_NAME      : CNAME
      NONUNIX_GROUP_NAME : CNAME

      %import common.CNAME
      %import common.WS
      %ignore "\\\\"
      %ignore WS
      %ignore /# comment/
```

Here's the input:

```
sudocfg = """
# comment
"""
```

The output is:

```
Traceback (most recent call last):
  File "./parse_sudoers.py", line 86, in <module>
    tree = sudo_parser.parse(sudocfg)
  File "/home/sgerasenko/proj/sudoers/lib/python2.7/site-packages/lark_parser-0.4.1-py2.7.egg/lark/lark.py", line 193, in parse
    return self.parser.parse(text)
  File "/home/sgerasenko/proj/sudoers/lib/python2.7/site-packages/lark_parser-0.4.1-py2.7.egg/lark/parser_frontends.py", line 144, in parse
    return self.parser.parse(text)
  File "/home/sgerasenko/proj/sudoers/lib/python2.7/site-packages/lark_parser-0.4.1-py2.7.egg/lark/parsers/xearley.py", line 128, in parse
    raise ParseError('Incomplete parse: Could not find a solution to input')
lark.common.ParseError: Incomplete parse: Could not find a solution to input
```

Ultimately my goal is to ignore everything from the beginning of the line to the end of the line though. I've tried `/^#[^\\n]+/`, but that didn't work either. I'm also not sure if the regex match is re.MULTILINE or not.

**Comment by erezsh** [comment]
Hi, it seems that there's a bug when the last token in the input is an %ignore token. I'll look into it soon.

**Comment by erezsh** [comment]
My guess is that this bug is also solved by the fix to issue #44 . Let me know if it solved it!

**Comment by erezsh** [comment]
It's been 3 weeks. I assume this issue is solved. Feel free to re-open it if it persists.
